### ISSUE TITLE
Missing concrete implementations of 'ModelProviderInterface.getModelTypeByModelName'

### ISSUE THREAD

**Issue by stouvoli** [aws-amplify/amplify-flutter#1455]
### Description

Hello,

How can I solve this issue. I got the error message: 

error: Missing concrete implementations of 'ModelProviderInterface.getModelTypeByModelName', 'getter ModelProviderInterface.customTypeSchemas', and 'setter ModelProviderInterface.customTypeSchemas'. (non_abstract_class_inherits_abstract_member at [numappdemo11032022] lib\models\ModelProvider.dart:21)

### Categories

- [ ] Analytics
- [ ] API (REST)
- [X] API (GraphQL)
- [ ] Auth
- [ ] Authenticator
- [ ] DataStore
- [ ] Storage

### Steps to Reproduce

I have folloed this steps https://docs.amplify.aws/lib/graphqlapi/getting-started/q/platform/flutter/#initialize-amplify-api.
1. Create a Flutter app
2. Init amplify and add api
3. Update dependencies with amplify_flutter: ^0.4.0 and amplify_api: ^0.4.0
4. Run amplify codegen models

### Screenshots

![image](https://user-images.githubusercontent.com/7300563/157954658-381077a7-31f3-44e0-a0cd-b04c6ad326cd.png)

### Platforms

- [ ] iOS
- [X] Android

### Environment

```bash
C:\Users\stouvoli\data\tools\flutter\bin\flutter.bat doctor --verbose
[√] Flutter (Channel stable, 2.10.3, on Microsoft Windows [Version 10.0.19044.1586], locale fr-CI)
    • Flutter version 2.10.3 at C:\Users\stouvoli\data\tools\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7e9793dee1 (9 days ago), 2022-03-02 11:23:12 -0600
    • Engine revision bd539267b4
    • Dart version 2.16.1
    • DevTools version 2.9.2

[√] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at C:\Users\stouvoli\AppData\Local\Android\sdk
    • Platform android-31, build-tools 32.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.11+9-b60-7590822)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.0.6)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.0.32126.317
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2021.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+9-b60-7590822)

[√] Android Studio (version 2020.3)
    • Android Studio at C:\Program Files\Android\Android Studio Preview\android-studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[√] IntelliJ IDEA Community Edition (version 2021.2)
    • IntelliJ at C:\Users\stouvoli\AppData\Local\JetBrains\Toolbox\apps\IDEA-C\ch-0\212.5457.46
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart

[√] Connected device (4 available)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86    • Android 11 (API 30) (emulator)
    • Windows (desktop)                  • windows       • windows-x64    • Microsoft Windows [Version 10.0.19044.1586]
    • Chrome (web)                       • chrome        • web-javascript • Google Chrome 99.0.4844.51
    • Edge (web)                         • edge          • web-javascript • Microsoft Edge 99.0.1150.36

[√] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
Process finished with exit code 0
```

### Dependencies

```bash
Dart SDK 2.16.1
Flutter SDK 2.10.3
numappdemo11032022 1.0.0+1

dependencies:
- amplify_api 0.4.1 [amplify_api_plugin_interface amplify_core collection flutter meta plugin_platform_interface]
- amplify_flutter 0.4.1 [amplify_analytics_plugin_interface amplify_api_plugin_interface amplify_auth_plugin_interface amplify_core amplify_datastore_plugin_interf
ace amplify_storage_plugin_interface collection flutter json_annotation meta plugin_platform_interface]
- cupertino_icons 1.0.4
- flutter 0.0.0 [characters collection material_color_utilities meta typed_data vector_math sky_engine]

transitive dependencies:
- amplify_analytics_plugin_interface 0.4.1 [amplify_core flutter meta]
- amplify_api_plugin_interface 0.4.1 [amplify_core collection flutter json_annotation meta]
- amplify_auth_plugin_interface 0.4.1 [flutter meta amplify_core]
- amplify_core 0.4.1 [flutter plugin_platform_interface collection date_time_format meta uuid]
- amplify_datastore_plugin_interface 0.4.1 [flutter meta collection amplify_core]
- amplify_storage_plugin_interface 0.4.1 [flutter meta amplify_core]
- characters 1.2.0
- collection 1.15.0
- crypto 3.0.1 [collection typed_data]
- date_time_format 2.0.1
- json_annotation 4.4.0 [meta]
- material_color_utilities 0.1.3
- meta 1.7.0
- plugin_platform_interface 2.1.2 [meta]
- sky_engine 0.0.99
- typed_data 1.3.0 [collection]
- uuid 3.0.6 [crypto]
- vector_math 2.1.1
```

### Device

Pixel_3a_API_30_x86

### OS

Android 11

### CLI Version

7.6.23

**Comment by ragingsquirrel3** [comment]
@stouvoli I have seen a similar issue related to codegen, especially with older versions of CLI. Do you mind sharing the code for your ModelProvider.dart after having run `amplify codegen models`? From your description seems like you have done this, but double-checking that you ran the codegen command with latest version of the amplify CLI and having successfully run `flutter pub get` in your project? Some of these fields in codegen depend on having recent amplify_flutter versions in pubpsec.lock to ensure compatibility.

**Comment by stouvoli** [comment]
C:\Users\stouvoli\data\Trainings\Exercices\AWS_Amplify\Flutter\numappdemo11032022>aws --version
aws-cli/2.2.18 Python/3.8.8 Windows/10 exe/AMD64 prompt/off

--------------------------------
C:\Users\stouvoli\data\Trainings\Exercices\AWS_Amplify\Flutter\numappdemo11032022>amplify codegen models

The following types do not have '@auth' enabled. Consider using @auth with @model
         - Todo
Learn more about @auth here: https://docs.amplify.aws/cli-legacy/graphql-transformer/auth

GraphQL schema compiled successfully.

Edit your schema at C:\Users\stouvoli\data\Trainings\Exercices\AWS_Amplify\Flutter\numappdemo11032022\amplify\backend\api\numappdemo11032022\schema.graphql or plac
e .graphql files in a directory at C:\Users\stouvoli\data\Trainings\Exercices\AWS_Amplify\Flutter\numappdemo11032022\amplify\backend\api\numappdemo11032022\schema

Detected Dart SDK version 2.12.0 or above
Detected feature flag: “enableDartNullSafety : true”
Generating Dart Models with null safety. To opt out of null safe models, turn off the “enableDartNullSafety” feature flag. Learn more: https://docs.amplify.aws/lib
/project-setup/null-safety/q/platform/flutter
Successfully generated models. Generated models can be found in C:\Users\stouvoli\data\Trainings\Exercices\AWS_Amplify\Flutter\numappdemo11032022\lib\models

C:\Users\stouvoli\data\Trainings\Exercices\AWS_Amplify\Flutter\numappdemo11032022>

---------------------
C:\Users\stouvoli\data\Trainings\Exercices\AWS_Amplify\Flutter\numappdemo11032022>flutter pub get
Running "flutter pub get" in numappdemo11032022...                 647ms


[ModelProvider.dart.txt](https://github.com/aws-amplify/amplify-flutter/files/8237443/ModelProvider.dart.txt)

**Comment by HuiSF** [comment]
- Run `amplify --version` to double check which version was installed
- Run `amplify uninstall -g @aws-amplify/cli` before install the latest version
- Make sure you don't have multiple versions of Amplify CLI installed in your system (e.g. global installation vs project installation)
- Run `npm list -g --depth=1` check the version of amplify-codegen it should be `2.28.0`

**Comment by stouvoli** [comment]
C:\Users\stouvoli\data\Trainings\Exercices\AWS_Amplify\Flutter\numappdemo11032022>amplify --version
7.6.23
The size of this file is suspect, 729MB!
I have run a search on my PC there is only one amplify.exe file.

![image](https://user-images.githubusercontent.com/7300563/158241816-4a56a7d6-fd8f-45d2-bc04-a6173ccc77fb.png)

C:\Users\stouvoli\data\Trainings\Exercices\AWS_Amplify\Flutter\numappdemo11032022>uninstall -g @aws-amplify/cli
'uninstall' is not recognized as an internal or external command,
operable program or batch file.

This command is not working. I use windows probably this is not a windows command.

I already have the amplify-codegen 2.28.0.
C:\Users\stouvoli\data\Trainings\Exercices\AWS_Amplify\Flutter\numappdemo11032022>npm list -g --depth=1
C:\Users\stouvoli\AppData\Roaming\npm
`-- @aws-amplify/cli@7.6.23
  +-- @aws-amplify/amplify-category-api@1.2.14
  +-- @aws-amplify/amplify-category-auth@2.7.4
  +-- @aws-amplify/amplify-category-custom@2.3.21
  +-- @aws-amplify/amplify-category-storage@3.1.18
  +-- @aws-amplify/amplify-util-uibuilder@1.2.15
  +-- @aws-amplify/graphql-auth-transformer@0.7.8
  +-- @aws-cdk/cloudformation-diff@1.124.0
  +-- amplify-app@4.2.23
  +-- amplify-category-analytics@3.2.22
  +-- amplify-category-function@3.5.3
  +-- amplify-category-geo@2.4.2
  +-- amplify-category-hosting@3.2.22
  +-- amplify-category-interactions@3.2.22
  +-- amplify-category-notifications@2.19.6
  +-- amplify-category-predictions@3.2.22
  +-- amplify-category-xr@3.2.22
  +-- amplify-cli-core@2.4.17
  +-- amplify-cli-logger@1.1.0
  +-- amplify-codegen@2.28.0

**Comment by stouvoli** [comment]
npm uninstall -g @aws-amplify/cli
Is working, unisntall in progress.

**Comment by stouvoli** [comment]
amplify uninstalled and reinstalled.
A new project created. I still have the same error message.

`error: Missing concrete implementations of 'ModelProviderInterface.getModelTypeByModelName', 'getter ModelProviderInterface.customTypeSchemas', and 'setter ModelProviderInterface.customTypeSchemas'. (non_abstract_class_inherits_abstract_member at [test14032022] lib\models\ModelProvider.dart:21)`

C:\Users\stouvoli\data\Trainings\Exercices\AWS_Amplify\Flutter\test14032022>amplify --version
7.6.23

C:\Users\stouvoli\data\Trainings\Exercices\AWS_Amplify\Flutter\test14032022>npm list -g --depth=1
C:\Users\stouvoli\AppData\Roaming\npm
`-- @aws-amplify/cli@7.6.23
  +-- @aws-amplify/amplify-category-api@1.2.14
  +-- @aws-amplify/amplify-category-auth@2.7.4
  +-- @aws-amplify/amplify-category-custom@2.3.21
  +-- @aws-amplify/amplify-category-storage@3.1.18
  +-- @aws-amplify/amplify-util-uibuilder@1.2.15
  +-- @aws-amplify/graphql-auth-transformer@0.7.8
  +-- @aws-cdk/cloudformation-diff@1.124.0
  +-- amplify-app@4.2.23
  +-- amplify-category-analytics@3.2.22
  +-- amplify-category-function@3.5.3
  +-- amplify-category-geo@2.4.2
  +-- amplify-category-hosting@3.2.22
  +-- amplify-category-interactions@3.2.22
  +-- amplify-category-notifications@2.19.6
  +-- amplify-category-predictions@3.2.22
  +-- amplify-category-xr@3.2.22
  +-- amplify-cli-core@2.4.17
  +-- amplify-cli-logger@1.1.0
  +-- amplify-codegen@2.28.0

**Comment by fjnoyp** [comment]
Hi @stouvali sorry you're having these issues.  

**tl;dr**

So please do the following: 

1) Delete your existing `ModelProvider.dart` file 
2) Run amplify codegen models

**Context:** 

The `ModelProvider.dart` file you've provided is simply missing implementations for the `customTypeSchemas` field which means your `amplify-codegen` was out of date. 

Based on what you've said above it seems you've uninstalled and reinstalled `amplify-codegen` but didn't actually rerun it ...

**Comment by stouvoli** [comment]
Hello @fjnoyp

File ModelProvider.dart deleted and amplify codegen models run.
Still the same error message.

C:\Users\stouvoli\data\Trainings\Exercices\AWS_Amplify\Flutter\test14032022>amplify codegen models

The following types do not have '@auth' enabled. Consider using @auth with @model
         - Todo
Learn more about @auth here: https://docs.amplify.aws/cli-legacy/graphql-transformer/auth

GraphQL schema compiled successfully.

Edit your schema at C:\Users\stouvoli\data\Trainings\Exercices\AWS_Amplify\Flutter\test14032022\amplify\backend\api\test14032022\schema.graphql or place .graphql f
iles in a directory at C:\Users\stouvoli\data\Trainings\Exercices\AWS_Amplify\Flutter\test14032022\amplify\backend\api\test14032022\schema

Detected Dart SDK version 2.12.0 or above
Detected feature flag: “enableDartNullSafety : true”
Generating Dart Models with null safety. To opt out of null safe models, turn off the “enableDartNullSafety” feature flag. Learn more: https://docs.amplify.aws/lib
/project-setup/null-safety/q/platform/flutter
Successfully generated models. Generated models can be found in C:\Users\stouvoli\data\Trainings\Exercices\AWS_Amplify\Flutter\test14032022\lib\models

![image](https://user-images.githubusercontent.com/7300563/158322626-c10b7c33-c02a-4673-977e-26ed7dca976d.png)

**Comment by stouvoli** [comment]
I have created another project and this time have followed this other tutorial https://docs.amplify.aws/start/getting-started/generate-model/q/integration/flutter/.

There is no error with the model created, when using "amplify pull --sandboxId ...".

It seems to be a problem with the command "amplify codegen models".

C:\Users\stouvoli\data\Trainings\Exercices\AWS_Amplify\Flutter\numapptest15032022>amplify pull --sandboxId 42a903be-13b6-47ce-9399-f6315f5bda26
✅ Amplify project setup for flutter platform
� Creating base Amplify project
✅ Successfully created base Amplify Project
✅ Amplify setup completed successfully.
Error: File at path: 'C:\Users\stouvoli\data\Trainings\Exercices\AWS_Amplify\Flutter\numapptest15032022\amplify\backend\amplify-meta.json' does not exist

Detected Dart SDK version 2.12.0 or above
Detected feature flag: “enableDartNullSafety : true”
Generating Dart Models with null safety. To opt out of null safe models, turn off the “enableDartNullSafety” feature flag. Learn more: https://docs.amplify.aws/lib
/project-setup/null-safety/q/platform/flutter
Successfully generated models. Generated models can be found in C:\Users\stouvoli\data\Trainings\Exercices\AWS_Amplify\Flutter\numapptest15032022\lib\models

![image](https://user-images.githubusercontent.com/7300563/158417068-5bed741b-ada3-45cc-a5d4-fa2927b60f1d.png)

**Comment by fjnoyp** [comment]
Hmm that is quite strange.  It appears that your `amplify codegen models` command is still generating an old version of the `ModelProvider` class without the required methods. 

It sounds like it's working for you now though? @stouvoli

**Comment by stouvoli** [comment]
Yes I'll move forward without using the command `amplify codegen models`.

**Comment by FireAndIceFrog** [comment]
Was this ever resolved?

**Comment by HuiSF** [comment]
Hi @FireAndIceFrog - The solution was to regenerated models using `amplify codegen models` with the latest version of Amplify CLI.

**Comment by a9udn9u** [comment]
I'm still getting this error, `@aws-amplify/cli` is at `9.2.1`, `amplify-codegen` does not appear to be a dependency of `@aws-amplify/cli` but I installed `amplify-codegen@3.0.5` just to make sure I didn't miss any packages.

**Comment by ainomegumi-desu** [comment]
still get this problem with `amplify/cli 10.5.1`  `amplify_flutter / 0.6.10 or 1.0.0-next.1+1`

**Comment by Abhishek-cmd13** [comment]
I am still facing the issue and I am clueless how to solve it , and the same issue is there for Query model Identifier.

**Comment by abdallahshaban557** [comment]
Hi @Abhishek-cmd13  - can you please create a new Github issue and provide us with the details of your used dependencies so that we can investigate it further?

**Comment by Sofianopoulou** [comment]
what worked for me is to add @model at the type of my schema.graphql, so that my schema.graphql looks like this:

**Comment by Giovanny-DS** [comment]
@Sofianopoulou you're the best!
